# HG changeset patch
# User Rik <rik@octave.org>
# Date 1407286416 25200
#      Tue Aug 05 17:53:36 2014 -0700
# Node ID b517b08d2e240d148fed7c0078f32fcf66cd4a19
# Parent  12d1297d3a380e34dca508dcb35fd06c455a17d4
# Parent  0b657f4e7780bfeffc65895c5dad7fe73aa37e8b
maint: Merge gui-release to default.

diff --git a/doc/interpreter/package.txi b/doc/interpreter/package.txi
--- a/doc/interpreter/package.txi
+++ b/doc/interpreter/package.txi
@@ -266,17 +266,17 @@ the package functions were installed.
 @item package/on_uninstall.m
 This is an optional function that is run prior to the removal of a
 package.  This function is called with a single argument, a struct with
 fields names after the data in the @file{DESCRIPTION}, the paths where
 the package functions are installed, and whether the package is currently
 loaded.
 @end table
 
-Besides the above mentioned files, a package can also contain on or
+Besides the above mentioned files, a package can also contain one or
 more of the following directories:
 
 @table @code
 @item package/inst
 An optional directory containing any files that are directly installed
 by the package.  Typically this will include any @code{m}-files.
 
 @item package/src
@@ -354,17 +354,17 @@ License: GPLv3+
 The package manager currently recognizes the following keywords
 
 @table @code
 @item Name
 Name of the package.
 
 @item Version
 Version of the package.  A package version must be 3 numbers separated
-by a dot.
+by dots.
 
 @item Date
 Date of last update.
 
 @item Author
 Original author of the package.
 
 @item Maintainer
diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -256,17 +256,31 @@ file_editor::call_custom_editor (const Q
   QSettings *settings = resource_manager::get_settings ();
 
   if (settings->value ("useCustomFileEditor").toBool ())
     {
       QString editor = settings->value ("customFileEditor").toString ();
       editor.replace ("%f", file_name);
       editor.replace ("%l", QString::number (line));
 
-      QProcess::startDetached (editor);
+      bool started_ok = QProcess::startDetached (editor);
+
+      if (started_ok != true)
+        {
+          QMessageBox *msgBox
+            = new QMessageBox (QMessageBox::Critical,
+                               tr ("Octave Editor"),
+                               tr ("Could not start custom file editor\n%1").
+                               arg (editor),
+                               QMessageBox::Ok, this);
+
+           msgBox->setWindowModality (Qt::NonModal);
+           msgBox->setAttribute (Qt::WA_DeleteOnClose);
+           msgBox->show ();
+        }
 
       if (line < 0 && ! file_name.isEmpty ())
         handle_mru_add_file (QFileInfo (file_name).canonicalFilePath ());
 
       return true;
     }
 
   return false;
diff --git a/scripts/special-matrix/toeplitz.m b/scripts/special-matrix/toeplitz.m
--- a/scripts/special-matrix/toeplitz.m
+++ b/scripts/special-matrix/toeplitz.m
@@ -69,17 +69,17 @@ function retval = toeplitz (c, r)
 
     r = c;
     nr = length (c);
     nc = nr;
   else
     if (! (isvector (c) && isvector (r))) 
       error ("toeplitz: C and R must be vectors");
     elseif (r(1) != c(1))
-      warning ("toeplitz: column wins anti-diagonal conflict");
+      warning ("toeplitz: column wins diagonal conflict");
     endif
 
     nr = length (c);
     nc = length (r);
   endif
 
   if (nr == 0 || nc == 0)
     ## Empty matrix.
